addons/binding/org.openhab.binding.icloud/ESH-INF/thing/thing-types.xml

```diff
@@ -58,6 +58,11 @@
 			<channel typeId="DeviceName" id="deviceName" />
 		</channels>
 
+        <config-description>
+           <parameter name="deviceId" type="text" required="true">
+               <label>Device id</label>

```

Comment:
ID?

Reply:
Yep; as in your example:

`Thing device myiPhone8 "iPhone 8" @ "World" [deviceId="VIRG9TsrvXfE90ewVBA1H5swtwEQePdXVjHq3Si6pdJY2Fjro8QlreHYVGSUzmWV"]`

I dropped the hash and use the id directly in the configuration; but as said - I do not know yet if it works. I´ll try to add the "representation-property". Do you know if that property needs to be present as channel?

Reply:
`name=deviceId` is perfectly, I wanted to ask if the label should be "Device ID"?

Sadly I can't say for sure but it seems like not more is needed. Please compare with the Sonos Binding as yet another example: https://github.com/eclipse/smarthome/pull/4035/files